src/scriptdev2/scripts/kalimdor/durotar.cpp

```diff
@@ -127,9 +127,10 @@ struct npc_lazy_peonAI : public ScriptedAI
                 }
                 case 2: //spawn
                 {
-                    m_creature->SetFacingTo(m_fSpawnO);
-                    m_creature->GetMotionMaster()->MoveTargetedHome(); //hacky way of getting the peon to sleep
+                    m_creature->GetMotionMaster()->MoveIdle();
                     m_creature->HandleEmote(EMOTE_STATE_NONE);
+                    m_creature->SetFacingTo(m_fSpawnO);
+                    m_creature->CastSpell((Unit*)m_creature, SPELL_PEON_SLEEP, false);
```

Comment:
in SD2 we don't use this format, unless the spell is missing from DBC and added by DB.
Please replace with ```DoCastSpellIfCan(m_creature, SPELL_PEON_SLEEP);```
Also the cast from m_creature to Unit* is not required, as m_creature is already a Unit.